modules/stochastic_tools/include/samplers/AIS.h

```diff
@@ -0,0 +1,80 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Sampler.h"
+#include "ReporterInterface.h"
+
+/**
+ * A class used to perform Adaptive Importance Sampling using a Markov Chain Monte Carlo algorithm
+ */
+
+class AIS : public Sampler, public ReporterInterface
+{
+public:
+  static InputParameters validParams();
+
+  AIS(const InputParameters & parameters);
+
+  std::vector<std::vector<Real>> _inputs_sto;
+
+protected:
+  /// Return the sample for the given row and column
+  virtual Real computeSample(dof_id_type row_index, dof_id_type col_index) override;
```

Comment:
add doxygen with description of parameters
I d like to understand why col_index = 0 is special

Reply:
The sampler cycles through all the parameters using col_index. For MCMC sampling, we need to propose all the next parameters at once. So, I am doing this proposal at col_index = 0 and storing these proposals. Then for col_index>0, I am simply returning the stored values. This is different from independent Monte Carlo where we can sample a parameter independently of the other parameters.